assets/js/plugin-check-admin.js

```diff
@@ -265,12 +266,28 @@
 
 		// Render remaining files with only warnings.
 		for ( const file in warnings ) {
+			errorWarningMessage = true;
 			renderFileResults( file, [], warnings[ file ] );
 		}
 
 		resultsContainer.innerHTML += renderTemplate(
 			'plugin-check-results-complete'
 		);
+
+		const resultsMessage = document.getElementById(
+			'plugin-check__message'
+		);
+		if ( resultsMessage ) {
+			const resultsMessageChild = resultsMessage.querySelector( 'p' );
+			if ( errorWarningMessage ) {
+				resultsMessage.classList.remove( 'notice-success' );
+				resultsMessage.classList.add( 'notice-error' );
+				resultsMessageChild.innerHTML += ' ' + pluginCheck.errorMessage;
+			} else {
+				resultsMessageChild.innerHTML +=
+					' ' + pluginCheck.successMessage;
+			}
+		}
```

Comment:
Instead of all this logic that manually tampers with the output of the `renderTemplate()` function, we should instead adjust that template itself (i.e. the `results-complete.php` file) to use placeholders (e.g. for the class and the additional sentence).

We should then pass the dynamic parts in the `data` parameter to the `renderTemplate()` function.

Reply:
@mukeshpanchal27 @felixarntz I agree here, lets make use of the template to set the message text and notice class depending on if there are errors or not.